src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirDataStore.cs

```diff
@@ -336,22 +336,42 @@ public async Task BulkUpdateSearchParameterIndicesAsync(IReadOnlyCollection<Reso
                     sqlCommandWrapper,
                     _bulkReindexResourcesTvpGeneratorVLatest.Generate(resources.ToList()));
 
-                try
+                if (_schemaInformation.Current >= SchemaVersionConstants.BulkReindexReturnsFailuresVersion)
                 {
-                    await sqlCommandWrapper.ExecuteNonQueryAsync(cancellationToken);
+                    // We will reindex the rest of the batch if one resource has a versioning conflict
+                    var failedResourceCount = (int?)await sqlCommandWrapper.ExecuteScalarAsync(cancellationToken);
```

Comment:
Can we still have exception during execution?

In else block we log it, and here we don't

Reply:
Ah thank you, I was focused on the exceptions and overlooked the logging. Put this in in 57a3a31.